### PR TITLE
Fix: make lint, cft tests happy

### PR DESCRIPTION
**Pull request by balajismaniam**

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -43,4 +43,6 @@ jobs:
             REGISTRY=$(grep "REGISTRY_URL := " $MAKEFILE | cut -d\  -f3)
             echo dev-tools=${REGISTRY}/${IMAGE}:${VERSION} >> "$GITHUB_OUTPUT"
           fi
-      - run: docker run --rm -v ${{ github.workspace }}:/workspace ${{ steps.variables.outputs.dev-tools }} /usr/local/bin/test_lint.sh
+      - run: docker run --rm -e EXCLUDE_LINT_DIRS -v ${{ github.workspace }}:/workspace ${{ steps.variables.outputs.dev-tools }} /usr/local/bin/test_lint.sh
```
Comment (bharathkkb) [inline .github/workflows/lint.yaml:46]: This file is managed centrally, please make edits in the main repo
https://github.com/GoogleCloudPlatform/terraform-genai-doc-summarization/blob/0ef4fa368e26c3b2a9f9ecac05f86d237bbb7373/.github/workflows/lint.yaml#L15-L16

File: test/setup/iam.tf [hunk 2]
```diff
@@ -16,8 +16,7 @@
 
 locals {
   int_required_roles = [
-    "roles/resourcemanager.projectIamAdmin",
-    "roles/storage.admin",
+    "roles/owner"
```
Comment (bharathkkb) [inline test/setup/iam.tf:19]: nit: let's add a todo to change this to granular roles

File: versions.tf [hunk 3]
```diff
@@ -19,16 +19,31 @@ terraform {
...
     module_name = "blueprints/terraform/gen-ai-document-summarization/v0.0.1"
   }
+}
 
-  backend "gcs" {
-    bucket  = null
-    prefix  = null
-  }
+provider "google" {
+  project = var.project_id
```
Comment (bharathkkb) [inline versions.tf:48]: Let's configure this per resource rather than configuring the provider

File: versions.tf [hunk 4]
```diff
@@ -19,16 +19,31 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = ">= 3.53, < 5.0"
+      version = ">= 4.66"
```
Comment (bharathkkb) [inline versions.tf:22]: It would be good to keep the `< 5.0` constraint as 5.0 may contain breaking changes.

**Review by bharathkkb — COMMENTED** [review]
some minor comments for TF

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (balajismaniam) [reply .github/workflows/lint.yaml:46]: I will keep this here for now to make lint checks happy. Will create a PR in CFT repo.

File: test/setup/iam.tf (hunk 2 above)
Comment (balajismaniam) [reply test/setup/iam.tf:20]: Done.

File: versions.tf (hunk 3 above)
Comment (balajismaniam) [reply versions.tf]: Thanks for catching this. Updated.

File: versions.tf (hunk 4 above)
Comment (balajismaniam) [reply versions.tf]: Done.

**Review by telpirion — APPROVED** [review]

**Pull request merged**